### PR TITLE
[#323] added Files.createTemp* Managed methods; added Files.deleteRec…

### PR DESCRIPTION
**Pull request by vitaliihonta**

### CODE REVIEW

File: nio/src/main/scala/zio/nio/file/Files.scala [hunk 1]
```diff
@@ -106,6 +133,9 @@ object Files {
   def deleteIfExists(path: Path): ZIO[Blocking, IOException, Boolean] =
     effectBlocking(JFiles.deleteIfExists(path.javaPath)).refineToOrDie[IOException]
 
+  def deleteRecursive(path: Path): ZIO[Blocking, IOException, Long] =
```
Comment (vitaliihonta) [inline nio/src/main/scala/zio/nio/file/Files.scala:136]: IMO this method may be useful independently of the rest updates

File: nio/src/main/scala/zio/nio/file/Files.scala [hunk 2]
```diff
@@ -82,13 +96,26 @@ object Files {
...
     fileAttributes: Iterable[FileAttribute[_]]
   ): ZIO[Blocking, IOException, Path] =
     effectBlocking(Path.fromJava(JFiles.createTempDirectory(prefix.orNull, fileAttributes.toSeq: _*)))
       .refineToOrDie[IOException]
 
+  def createTempDirectoryManaged(
+    prefix: Option[String],
+    fileAttributes: Iterable[FileAttribute[_]]
+  ): ZManaged[Blocking, IOException, Path] =
+    ZManaged.make(createTempDirectory(prefix, fileAttributes))(release = deleteRecursive(_).ignore)
```
Comment (vitaliihonta) [inline nio/src/main/scala/zio/nio/file/Files.scala:117]: Not 100% sure about recursive directory deletion. But it seems to be reasonable in terms of temporary directories

File: nio/src/main/scala/zio/nio/file/Files.scala [hunk 3]
```diff
@@ -74,6 +81,13 @@ object Files {
     effectBlocking(Path.fromJava(JFiles.createTempFile(prefix.orNull, suffix, fileAttributes.toSeq: _*)))
       .refineToOrDie[IOException]
 
+  def createTempFileManaged(
+    suffix: String = ".tmp",
+    prefix: Option[String],
+    fileAttributes: Iterable[FileAttribute[_]]
+  ): ZManaged[Blocking, IOException, Path] =
+    ZManaged.make(createTempFile(suffix, prefix, fileAttributes))(release = deleteIfExists(_).ignore)
```
Comment (vitaliihonta) [inline nio/src/main/scala/zio/nio/file/Files.scala:89]: Not sure what exactly should be used in the finalization block: ignore or orDie?
In most cases, it won't fail so I choose to ignore

Comment (quelgar) [reply nio/src/main/scala/zio/nio/file/Files.scala:89]: Yeah I'm not 100% sure either, but `ignore` seems good.

File: nio/src/main/scala/zio/nio/file/Files.scala [hunk 4]
```diff
@@ -74,6 +81,13 @@ object Files {
     effectBlocking(Path.fromJava(JFiles.createTempFile(prefix.orNull, suffix, fileAttributes.toSeq: _*)))
       .refineToOrDie[IOException]
 
+  def createTempFileManaged(
+    suffix: String = ".tmp",
+    prefix: Option[String],
```
Comment (quelgar) [inline nio/src/main/scala/zio/nio/file/Files.scala:86]: Should be fine to have `prefix` default to `None` and `fileAttributes` to default to `Iterable.empty`?

File: nio/src/test/scala/zio/nio/file/FilesSpec.scala [hunk 5]
```diff
@@ -0,0 +1,92 @@
+package zio.nio.file
+
+import zio.{ Chunk, Ref }
+import zio.nio.BaseSpec
+import zio.nio.core.file.Path
+import zio.test._
+import zio.test.Assertion._
+
+object FilesSpec extends BaseSpec {
```
Comment (quelgar) [inline nio/src/test/scala/zio/nio/file/FilesSpec.scala:9]: 🚀

**Review by quelgar — APPROVED** [review]
Great, thank you!

**Pull request merged**